### ISSUE TITLE
代码无法训练

### ISSUE THREAD

**Issue by aa12356jm** [THUFutureLab/gluon-face#11]
昨天使用代码来跑一下模型试试，结果发现一直卡在 train_script.py中的这一行batch = next(batch_generator)，我用的是4个v100，显存占用是正常的，但是gpu利用率一直为零，应该是一直卡在数据加载这一块。
单步调试发现在函数中
```
def inf_train_gen(loader):
    while True:
        for batch in loader:
            yield batch       
```      
**执行完这一句for batch in loader:，就一直卡在这里了**

**环境如下：**
ubuntu16.04
mxnet版本是1.5.0

**代码只修改了如下2处：**
（1）将~/.mxnet/datasets/face的所有路径替换为自己的数据路径
（2）mobilefacenet-arcface.sh内容修改如下：
`python train_script.py --dataset emore --batch-size 1760 --ctx 0,1,2,3 --dtype float32 -j 24 --lr 0.35 --lr-mode cosine --lr-warmup-iters 5000 --wd 0.00004 --no-wd -s 64 -m 0.5 -t lfw,agedb_30 -n mobilefacenet --niters 86000 --loss arcface --loss-warmup-iters 35000 --hybrid --cat-interval 500`

**Comment by DevinXYang** [comment]
你替换自己数据的话除非数据格式一模一样不然要重写Dataset类

**Comment by aa12356jm** [comment]
数据还是emore的数据，只是数据存储路径我改变了，其他都一样的

**Comment by DevinXYang** [comment]
用的是Python3吗?

**Comment by aa12356jm** [comment]
是python3.6

**Comment by aa12356jm** [comment]
会不会是mxnet的版本问题？你们用的是哪个版本？

**Comment by DevinXYang** [comment]
你单独测一下Dataset和Dataloader能不能正常工作.
```python
dt = dataset()
print(dt[0])
```
dataloader类似

**Comment by DevinXYang** [comment]
换路径的话检查一下face下的路径格式对不对,参考README.md

**Comment by aa12356jm** [comment]
路径格式没问题，单独测试了一下Dataset和Dataloader，Dataset可以，Dataloader不行，感觉是Dataloader中的多线程加载的问题

**Comment by aa12356jm** [comment]
把DataLoader()的参数num_workers=0就可以使用了，这个是mxnet的bug吗？

**Comment by DevinXYang** [comment]
Dataloader的话看看是不是设置的线程超过CPU线程数了？

**Comment by aa12356jm** [comment]
我们的服务器是DGX-1 ,线程数设置了12,24,48，都不行，不知道咋回事，就用单线程跑跑看吧

**Comment by DevinXYang** [comment]
用htop看看CPU线程数，如果没有超过线程数也不行的话可以去mxnet的repo看看相关issues。

**Comment by aa12356jm** [comment]
htop显示的是80个线程

**Comment by aa12356jm** [comment]
我们是在docker中跑的，有可能是这个问题。。。

**Comment by DevinXYang** [comment]
docker中我们也遇到过这个问题，解决方案也是把worker设成0原因未知。

**Comment by aa12356jm** [comment]
嗯，在docker中设置num_workers=0就可以了

**Comment by DevinXYang** [comment]
Try to use `--shm-size "4g"` to build docker image.
